### PR TITLE
Danielh/eng 833 coalesce multiple opsani dev remedies to

### PR DESCRIPTION
**Pull request by DanielHHowell**

### CODE REVIEW

**Review by linkous8 — CHANGES_REQUESTED** [review]
from a cursory glance a couple of items:
* I dont see any automated unit/integration tests added that cover both types of the configuration
* You added a CLI flag for the checks which is good but I thought there was some kind of check configuration section for the servo.yaml. Can we get it added there as well as it may be required for environment variable configuration

**Comment by DanielHHowell** [comment]
> from a cursory glance a couple of items:
> 
> * I dont see any automated unit/integration tests added that cover both types of the configuration
> * You added a CLI flag for the checks which is good but I thought there was some kind of check configuration section for the servo.yaml. Can we get it added there as well as it may be required for environment variable configuration

@linkous8 I haven't found a check configuration outside of the CLI flag. Also I don't see any tests that cover the flags of the relevant CLI section, will see about adding that. Do you think an integration test running the two variants of checking on say, fiber-http would be good?

**Comment by linkous8** [comment]
> @linkous8 I haven't found a check configuration outside of the CLI flag.

Lets go ahead and add a new ChecksConfiguration model for that, say, around here: https://github.com/opsani/servox/blob/d027184c45e0bd2e15724a7a4c9af336602cf57a/servo/configuration.py#L467

Long term, I'm looking to streamline the CLI entrypoint to make the flow more understandable and will be preferring file/env var based configuration over CLI flags as the latter is more difficult to keep track of and alter programmatically

> Also I don't see any tests that cover the flags of the relevant CLI section, will see about adding that. Do you think an integration test running the two variants of checking on say, fiber-http would be good?

See if you can adapt this test to test both variants of the configuration: https://github.com/opsani/servox/blob/d027184c45e0bd2e15724a7a4c9af336602cf57a/tests/connectors/opsani_dev_test.py#L1176

File: servo/checks.py [hunk 1]
```diff
@@ -735,6 +737,152 @@ class Config:
...
+    @classmethod
+    async def process_checks(
+        cls,
+        checks_config: servo.configuration.ChecksConfiguration,
+        results: list,
+        passing: set,
+    ) -> bool:
+
+        remedy = checks_config.remedy
+        check_halting = checks_config.check_halting
```
Comment (linkous8) [inline servo/checks.py:750]: These variables are used once each, please reference the config directly for brevity and clarity on lines 791 and 799

File: servo/checks.py [hunk 2]
```diff
@@ -735,6 +737,152 @@ class Config:
         extra = pydantic.Extra.allow
 
 
+class CheckHelpers(pydantic.BaseModel, servo.logging.Mixin):
+    @classmethod
+    async def process_checks(
+        cls,
+        checks_config: servo.configuration.ChecksConfiguration,
+        results: list,
+        passing: set,
```
Comment (linkous8) [inline servo/checks.py:746]:
```suggestion
        results: list[servo.events.EventResult],
        passing: set[str],
```

File: servo/checks.py [hunk 3]
```diff
@@ -31,6 +32,7 @@
 )
 
 import pydantic
+from tabulate import tabulate
 
 import servo.configuration
 import servo.logging
```
Comment (linkous8) [inline servo/checks.py:38]: add import for servo.events (note the event reduction may be better located in the servo if events import turns into a circular reference anywhere)

File: servo/checks.py [hunk 4]
```diff
@@ -735,6 +737,152 @@ class Config:
...
+
+        remedy = checks_config.remedy
+        check_halting = checks_config.check_halting
+
+        ready = False
+        failure = None
+
+        checks: List[servo.Check] = functools.reduce(
+            lambda a, b: a + b.value, results, []
+        )
```
Comment (linkous8) [inline servo/checks.py:757]:
```suggestion
        checks: list[Check] = functools.reduce(lambda a, b: a + b.value, results, [])
```

File: servo/checks.py [hunk 5]
```diff
@@ -735,6 +737,152 @@ class Config:
...
+                    break
+
+        if not failure:
+            servo.logger.info("🔥 All checks passed.")
+            ready = True
+
+        return ready
+
+    @classmethod
+    async def checks_to_table(cls, checks_config, results) -> Tuple[bool, str]:
```
Comment (linkous8) [inline servo/checks.py:809]:
```suggestion
    async def checks_to_table(cls, checks_config, results) -> str:
```

File: servo/checks.py [hunk 6]
```diff
@@ -735,6 +737,152 @@ class Config:
...
+                message = functools.reduce(
+                    lambda m, e: m + f"({errors.index(e) + 1}/{len(errors)}) {e}\n",
+                    errors,
+                    "",
+                )
+                row = [result.connector.name, status, message]
+                table.append(row)
+
+        # Output table
+        if not quiet:
```
Comment (linkous8) [inline servo/checks.py:879]: Move this check to line 476 of servo.py. to_table should always return a table

**Review by linkous8 — APPROVED** [review]
A few minor nitpicks aside, LGTM! 🚀

File: servo/servo.py [hunk 7]
```diff
@@ -403,6 +402,102 @@ def top_level_schema_json(self, *, all: bool = False) -> str:
...
+                checks_config=self.config.checks,
+                results=results,
+                passing=passing,
+            )
+            if not progressive:
+                output = await servo.checks.CheckHelpers.checks_to_table(
+                    checks_config=self.config.checks, results=results
+                )
+                if not quiet:
+                    print_callback(output)
```
Comment (linkous8) [inline servo/servo.py:482]:
```suggestion
            if not progressive and not quiet:
                output = await servo.checks.CheckHelpers.checks_to_table(
                    checks_config=self.config.checks, results=results
                )
                print_callback(output)
```

Comment (linkous8) [reply servo/servo.py:482]: No need to process table if quiet

**Pull request merged**